python/activator/middleware_interface.py

```diff
@@ -75,22 +72,35 @@ class MiddlewareInterface:
         appropriate for use in the Google Cloud environment; typically only
         change this when running local tests.
     """
+
     def __init__(self, central_butler: Butler, image_bucket: str, instrument: str,
                  butler: Butler,
-                 prefix: str = "gs://"):
+                 prefix: str = "s3://"):
         self.prefix = prefix
         self.central_butler = central_butler
         self.image_bucket = image_bucket
         self.instrument = lsst.obs.base.utils.getInstrument(instrument)
 
+        self.calibration_collection = f"{self.instrument.getName()}/calib"
+        self.output_collection = f"{self.instrument.getName()}/prompt"
+
         self._init_local_butler(butler)
         self._init_ingester()
         # TODO DM-34098: note that we currently need to supply instrument here.
         self.camera = self.central_butler.get("camera", instrument=self.instrument.getName())
         self.skymap = self.central_butler.get("skyMap")
 
-        # self.r = self.src.registry
-        self.calibration_collection = f"{instrument}/calib"
+        # TODO: we probably want to be able to configure this per-instrument?
+        # TODO: would be nice to remap ap_pipe/pipelines to use getName convention
+        camera_paths = {"DECam": "DarkEnergyCamera", "HSC": "HyperSuprimeCam"}
+        try:
+            camera_path = camera_paths[self.instrument.getName()]
+        except KeyError:
+            raise NotImplementedError("No ApPipe.yaml defined for camera {}" % self.instrument.getName())
+        ap_pipeline_file = os.path.join(lsst.utils.getPackageDir('ap_pipe'),
+                                        "pipelines/{}/ApPipe.yaml".format(camera_path))
+        self.pipeline = lsst.pipe.base.Pipeline.fromFile(ap_pipeline_file)
```

Comment:
I don't like keeping the PIPELINE_MAP approach, since we don't know why a given pipeline would be chosen right now (i.e. until we know what the trigger for switching pipelines is, it's hard to pick how to make the switch). Maybe we could move this pipeline definition into `prep_butler`, since if we do switch pipelines, it would probably happen in there?

Reply:
@ktlim assumed it would be keyed to the `kind` field in the `next_visit` message. I think sticking to that assumption is good enough for now, since it means that `MiddlewareInterface` is not responsible for the decision.

Reply:
Do we have any facility for different "kind"s of pipelines in ap_pipe?

Reply:
They would be just be different `ap_pipe/pipelines/.yaml` pipelines.

Reply:
All I'm asking is to not remove the support that was already implemented.

Reply:
I think what I'm trying to say is that it's not "support" if it doesn't actually do anything right now (we don't have and don't want ap pipelines for calib files), and doesn't look like what we'll actually need in the end (we need to figure out what the ap_pipe part of this will look like, too).

Reply:
Please do not try to redesign `ap_pipe` based on implementation details of `MiddlewareInterface`! It's very hard to maintain code where every component needs to know everything about what every other component does.

Reply:
I agree completely! I'm not saying we should redesign ap_pipe at all. I'm saying we'll have to think about that when it comes time to implement these things, and what was in here previously is not what we actually want.

Reply:
What I was thinking of was that `kind` would trigger different pipeline YAML files (e.g. ones in https://github.com/lsst/cp_pipe/tree/main/pipelines) as well as possibly different code in `prep_butler()`.  You shouldn't assume that everything being run by this framework is AP-written.

Reply:
Yes, but we need to decide how we'll actually go about that decision process (e.g. if we're trying to load packages from elsewhere, what other packages need to go into that dictionary?). Just saying "there will be a dict" seems too simple and misleading.